src/OpenFeatureSDK/OpenFeature.cs

```diff
@@ -30,19 +36,41 @@ public sealed class OpenFeature
         /// Sets the feature provider
         /// </summary>
         /// <param name="featureProvider">Implementation of <see cref="FeatureProvider"/></param>
-        public void SetProvider(FeatureProvider featureProvider) => this._featureProvider = featureProvider;
+        public void SetProvider(FeatureProvider featureProvider)
+        {
+            this._featureProviderLock.EnterWriteLock();
+            try

```

Comment:
None of these code blocks can throw, but when using this type of lock I think using try/finally is a customary way to release the lock.

Reply:
Yep, agreed. In fact, I made a little lock-wrapper [class](https://github.com/open-feature/java-sdk/pull/131/files#diff-f7f996e84ae9d77890aa5aa0cde5a7e3b08401b43173268bc134083536ea26dc) that allows us to use a try-with-resources block in Java (I think c# has an equivalent with the `using` keyword and `IDisposable` implementations).

Reply:
Especially for double locks, where they could nest the using statement.